OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/19.3.0/Dockerfile

```diff
@@ -85,62 +107,95 @@ ENV SETUP_LINUX_FILE="setupLinuxEnv.sh" \
 
 # Setup Scripts
-COPY $RUN_FILE $ADDNODE $ADDNODE_RSP $SETUPSSH $FUNCTIONS $CONFIGGRID $GRID_INSTALL_RSP $DBCA_RSP $PWD_FILE $CHECK_DB_FILE $USER_SCRIPTS_FILE $STOP_FILE $CHECK_DB_FILE $REMOTE_LISTENER_FILE $SETUPGRIDENV $DELNODE $RESET_OS_PASSWORD $MULTI_NODE_INSTALL  $SCRIPT_DIR/
+COPY $RUN_FILE $INITSH $ADDNODE $ADDNODE_RSP $SETUPSSH $FUNCTIONS $CONFIGGRID $GRID_INSTALL_RSP $DBCA_RSP $PWD_FILE $CHECK_DB_FILE $USER_SCRIPTS_FILE $STOP_FILE $CHECK_DB_FILE $REMOTE_LISTENER_FILE $SETUPGRIDENV $DELNODE $RESET_OS_PASSWORD $MULTI_NODE_INSTALL  $SCRIPT_DIR/
 
 RUN chmod 755 $INSTALL_SCRIPTS/*.sh  && \
     sync && \
     $INSTALL_DIR/install/$CHECK_SPACE_FILE && \
     $INSTALL_DIR/install/$SETUP_LINUX_FILE && \
     $INSTALL_DIR/install/$GRID_SETUP_FILE && \
     $INSTALL_DIR/install/$DB_SETUP_FILE && \
-    sed -e '/hard *memlock/s/^/#/g' -i /etc/security/limits.d/oracle-database-preinstall-19c.conf && \
-    sed -e '/ *nofile /s/^/#/g' -i /etc/security/limits.d/oracle-database-preinstall-19c.conf && \
-    su  $GRID_USER -c "$INSTALL_DIR/install/$INSTALL_GRID_BINARIES_FILE EE $PATCH_NUMBER" && \
-    $INVENTORY/orainstRoot.sh && \
-    $GRID_HOME/root.sh && \
-    su  $DB_USER  -c  "$INSTALL_DIR/install/$INSTALL_DB_BINARIES_FILE EE" && \
-    su  $DB_USER  -c  "$INSTALL_DIR/install/$ENABLE_RAC_FILE" && \
-    $INVENTORY/orainstRoot.sh && \
-    $DB_HOME/root.sh && \
-    su  $GRID_USER -c "$INSTALL_SCRIPTS/$GRID_HOME_CLEANUP" && \
-    su  $DB_USER -c "$INSTALL_SCRIPTS/$ORACLE_HOME_CLEANUP" && \
+    sync
+
+#############################################
+# -------------------------------------------
+# Start new stage for installing the grid and DB
+# -------------------------------------------
+#############################################
+
+from base AS builder
+
+COPY $GRID_SW_INSTALL_RSP  $INSTALL_GRID_PATCH $INSTALL_GRID_BINARIES_FILE $DB_INSTALL_RSP $INSTALL_DB_BINARIES_FILE $ENABLE_RAC_FILE $GRID_HOME_CLEANUP $ORACLE_HOME_CLEANUP $INSTALL_FILE_1 $INSTALL_FILE_2 $INSTALL_SCRIPTS/
+
+RUN chmod 755 $INSTALL_SCRIPTS/*.sh
+
+## Install software if SLIMMING is false
+RUN if [ "${SLIMMING}x" != 'truex' ]; then \
+      sed -e '/hard *memlock/s/^/#/g' -i /etc/security/limits.d/oracle-database-preinstall-19c.conf && \
+      sed -e '/ *nofile /s/^/#/g' -i /etc/security/limits.d/oracle-database-preinstall-19c.conf && \
+      su  $GRID_USER -c "$INSTALL_DIR/install/$INSTALL_GRID_BINARIES_FILE EE $PATCH_NUMBER" && \
+      $INVENTORY/orainstRoot.sh && \
+      $GRID_HOME/root.sh && \
+      su  $DB_USER  -c  "$INSTALL_DIR/install/$INSTALL_DB_BINARIES_FILE EE" && \
+      su  $DB_USER  -c  "$INSTALL_DIR/install/$ENABLE_RAC_FILE" && \
+      $INVENTORY/orainstRoot.sh && \
+      $DB_HOME/root.sh && \
+      su  $GRID_USER -c "$INSTALL_SCRIPTS/$GRID_HOME_CLEANUP" && \
+      su  $DB_USER -c "$INSTALL_SCRIPTS/$ORACLE_HOME_CLEANUP" && \
+      :; \
+    fi
+
+RUN rm -f $INSTALL_DIR/install/* && \
+    sync
```

Comment:
Shouldn't all three of these RUN directives be combined so that if `$SLIMMING` is not true, it still only uses a single layer?

Reply:
At this moment, I just added SLIMMING option but haven't published in README.md.  We did the testing based on uploaded dockerfile. I will do more testing and will change it later based on your suggestions.